src/platform/GenericOTARequestorDriver.cpp

```diff
@@ -70,14 +70,14 @@ void GenericOTARequestorDriver::UpdateDownloaded()
 void GenericOTARequestorDriver::UpdateConfirmed(System::Clock::Seconds32 delay)
 {
     VerifyOrDie(mImageProcessor != nullptr);
-    ScheduleDelayedAction(OTAUpdateStateEnum::kDelayedOnApply, delay,
+    ScheduleDelayedAction(UpdateFailureState::kApplying, delay,
                           [](System::Layer *, void * context) { ToDriver(context)->mImageProcessor->Apply(); });
 }
 
 void GenericOTARequestorDriver::UpdateSuspended(System::Clock::Seconds32 delay)
 {
     VerifyOrDie(mRequestor != nullptr);
-    ScheduleDelayedAction(OTAUpdateStateEnum::kDelayedOnApply, delay,
+    ScheduleDelayedAction(UpdateFailureState::kAwaitingNextAction, delay,
```

Comment:
```suggestion
    ScheduleDelayedAction(UpdateFailureState::kApplying, delay,
```
? Or was it intentional?

Reply:
Yes, this is intentional. I plan to handle different error states in `HandleError` in a different PR and wanted to make sure I differentiate these cases (here, if `UpdateSuspsended` is called, we did get a status of `AwaitNextAction` from the provider so we should retry or something. I suspect I'll need to rework some of these states as I make that PR.

Reply:
Ok, gotcha :)